### ISSUE TITLE
Browser build debug stack no longer works

### ISSUE THREAD

**Issue by pyjsorg** [pyjs/original-pyjs#169]
<b>What steps will reproduce the problem?</b>
1. Introduce a coding error on purpose to a module from an example
2. Build the example with the &quot;-d&quot; debug flag
3. Run the example - when the error is encountered, instead of a readable stack, you see more errors.

as an example, I introduced a simple error in the KitcehnSink example file &quot;Popup.py&quot; :

--- a/examples/kitchensink/Popups.py
+++ b/examples/kitchensink/Popups.py
@@ -38,7 +38,7 @@ class Popups(Sink):
             p = MyPopup()
             left = sender.getAbsoluteLeft() + 10
             top = sender.getAbsoluteTop() + 10
-            p.setPopupPosition(left, top)
-            p.setPopupPosition(left, to)
           p.show()
       elif sender == self.fDialogButton:
           dlg = MyDialog(self.baseURL())

After building with &quot;-d&quot; and running the example. When the popup is activated, this is what appears in the browser console:

unsupported operand type(s) for -: 'None', '0' Traceback:
Kitche...he.html (line 38663)

TypeError: $pyjs.loaded_modules[stack.module] is undefined Traceback:
Kitche...he.html (line 38663)

$pyjs.loaded_modules[stack.module] is undefined
[Break On This Error] msg = $pyjs.loaded_modules[stack.module]['**track_lines**'][stack['lineno']];
Kitche...he.html (line 17253)
uncaught exception: TypeError: unsupported operand type(s) for -: 'None', '0'

_Original issue: http://code.google.com/p/pyjamas/issues/detail?id=550 (February 15, 2011 02:58:06)_

**Comment by pyjsorg** [comment]
_From rich.new...@gmail.com on February 19, 2011 04:42:11:_
The error was introduced by one in a series of commits made by Kees last November:

Author: Kees Bos cornelis.bos@gmail.com  2010-11-18 06:45:04
Committer: Kees Bos cornelis.bos@gmail.com  2010-11-18 06:52:44
Parent: e5d1c443107199146036d35f76099abf7f41d2d7 (Support for late defned vairable/functions)
Branches: master, remotes/origin/master
Follows:
Precedes:

```
Remove creation of local vars with module, since it pollutes the variable stack
```

The exception stack is correct immediately before this commit, and from this commit until today it is not correct or not available at all.

**Comment by pyjsorg** [comment]
_From cornelis...@gmail.com on February 19, 2011 11:20:58:_
Thanks. I'll have a look. Since I broke it, I should be able to repair it...

**Comment by pyjsorg** [comment]
_From cornelis...@gmail.com on February 19, 2011 12:04:14:_
Well, the problem on a different commit.

$ git bisect good
85fb8f04d4d3a9a091da6f90e69d4f0f3c6487ee is first bad commit
commit 85fb8f04d4d3a9a091da6f90e69d4f0f3c6487ee
Author: Waldemar Kornewald <wkornewald>
Date:   Fri Nov 26 18:00:43 2010 +0100

```
Fixed sys.exc_info() to only return the traceback starting from the current context (this allows the code to work with Python's native unittest module)
```

:040000 040000 98dd7f80f03d32f37e135662a9aa8035067cd8e4 16c3d6ac6b5b5db85f857e5a489891c652d1dd90 M  pyjs

I've used the following patch to reproduce/look into the problem:
$ git diff
diff --git a/examples/kitchensink/Popups.py b/examples/kitchensink/Popups.py
index e407d84..9334ac3 100644
--- a/examples/kitchensink/Popups.py
+++ b/examples/kitchensink/Popups.py
@@ -38,7 +38,9 @@ class Popups(Sink):
             p = MyPopup()
             left = sender.getAbsoluteLeft() + 10
             top = sender.getAbsoluteTop() + 10
-            p.setPopupPosition(left, top)
-            from **pyjamas** import debugger
-            debugger()
-            p.setPopupPosition(left, to)
           p.show()
       elif sender == self.fDialogButton:
           dlg = MyDialog(self.baseURL())

**Comment by pyjsorg** [comment]
_From cornelis...@gmail.com on February 21, 2011 07:50:22:_
I think I fixed it. Could you check?

**Comment by pyjsorg** [comment]
_From rich.new...@gmail.com on February 21, 2011 22:18:08:_
THIS PROBLEM IS FIXED by Kees' commit dbee9b1a2d07234ff1d0d929ff922

This issue can be closed
